src/lib/legacy/Zikula/View.php

```diff
@@ -203,7 +203,7 @@ public function __construct(Zikula_ServiceManager $serviceManager, $moduleName =
         // the topmost module. For Hooks, Blocks, API Functions and others
         // you need to set this property to the name of the respective module!
         $masterRequest = \ServiceUtil::get('request_stack')->getMasterRequest();
-        $masterRequestModule = $masterRequest->attributes->has('_zkModule') ? $masterRequest->attributes->get('_zkModule') : null;
+        $masterRequestModule = !empty($routeParams['_zkModule']) ? $routeParams['_zkModule'] : null;
```

Comment:
this is not related to sessions, but something else I was working on related to #2935 (but does not fix it)